Source/WebCore/html/parser/HTMLDocumentParserFastPath.cpp

```diff
@@ -464,6 +464,7 @@ class HTMLFastPathParser {
 
     template<typename ParentTag> void parseCompleteInput()
     {
+        skipWhile<isASCIIWhitespace>(m_parsingBuffer);
```

Comment:
The trimming likely needs to happen in parseFromString(), just before calling the fast parser, not in the fast parser, since this behavior is specific to parseFromString().